pkg/metricprovider/aks/trackers/trackers.go

```diff
@@ -0,0 +1,543 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package trackers feature implementations that help track specific stats about
+// Kubernetes resources, e.g., nodes and pods in the AKS metric provider.
+package trackers
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+const (
+	// AKSClusterNodeSKULabelName is the node label added by AKS, which indicated the SKU
+	// of the node.
+	AKSClusterNodeSKULabelName = "beta.kubernetes.io/instance-type"
+)
+
+// supportedResourceNames is a list of resource names that the AKS metric provider supports.
+//
+// Currently the supported resources are CPU and memory.
+var supportedResourceNames []corev1.ResourceName = []corev1.ResourceName{
+	corev1.ResourceCPU,
+	corev1.ResourceMemory,
+}
+
+// NodeSet is a set of nodes.
+type NodeSet map[string]bool
+
+// NodeTracker helps track specific stats about nodes in a Kubernetes cluster, e.g., its count.
+type NodeTracker struct {
+	// totalCapacity and totalAllocatable are the total capacity and allocatable capacity
+	// of the cluster, respectively.
+	totalCapacity    corev1.ResourceList
+	totalAllocatable corev1.ResourceList
```

Comment:
do we compute the available on the fly?

Reply:
Hi Ryan! There's a separate tracker for pods that tallies the requested resources; when the available capacity info is requested, the provider simply subtracts one tally from another.